QuickFIXn/DataDictionary/DataDictionary.cs

```diff
@@ -574,65 +591,73 @@ private void ParseMsgEl(XmlNode node, DDMap ddmap, bool? componentRequired)
                 Console.WriteLine(s);
                 */
 
-                var fieldName = (childNode.Attributes["name"] != null) ? childNode.Attributes["name"].Value : "no-name";
+                VerifyChildNode(childNode, node);
 
-                if (childNode.Name == "field")
+                var nameAttribute = childNode.Attributes["name"].Value;
+                bool required = childNode.Attributes["required"]?.Value == "Y";
+
+                switch(childNode.Name)
                 {
-                    if (FieldsByName.ContainsKey(fieldName) == false)
-                    {
-                        throw new Exception(
-                            $"Field '{fieldName}' is used in '{messageTypeName}', but is not defined in <fields> set.");
-                    }
+                    case "field":
+                        if (FieldsByName.ContainsKey(nameAttribute) == false)
```

Comment:
```suggestion
                        if (!FieldsByName.ContainsKey(nameAttribute))
```

Reply:
personal preference: I like the more visually-apparent `== false`